File: src/main/java/br/com/blz/testjava/controller/dto/ProductDTO.java
package br.com.blz.testjava.controller.dto;

import java.util.Optional;

public class ProductDTO {
    private Long sku;
    private String name;
    private InventoryDTO inventory = new InventoryDTO();

    public Long getSku() {
        return sku;
    }

    public void setSku(Long sku) {
        this.sku = sku;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public InventoryDTO getInventory() {
        return inventory;
    }

    public void setInventory(InventoryDTO inventory) {
        this.inventory = inventory;
    }

    public Boolean getIsMarketable() {
        final Integer quantity = Optional.ofNullable(inventory)
            .orElse(new InventoryDTO())
            .getQuantity();

        return quantity.intValue() > 0;
    }
}

File: src/main/kotlin/br/com/blz/testjava/controller/dto/ProductDTO.kt
package br.com.blz.testjava.controller.dto

import br.com.blz.testjava.domain.entities.Inventory
import br.com.blz.testjava.domain.entities.Product
import br.com.blz.testjava.domain.entities.Warehouse
import javax.validation.constraints.NotEmpty
import javax.validation.constraints.NotNull

class ProductDTO() {
  @NotNull
  var sku: Long? = null
  @NotEmpty
  var name: String = ""
  var inventory: InventoryDTO = InventoryDTO()
  var isMarketable: Boolean = false

  constructor(product: Product): this() {
    this.sku = product.sku
    this.name = product.name
    this.inventory = InventoryDTO(product.inventory)
    this.isMarketable = product.isMarketable()
  }

  fun toEntity(): Product {
    return Product(sku!!, name,
             Inventory(inventory.warehouses
                    .map { warehouseDTO ->
                          Warehouse(warehouseDTO.locality, warehouseDTO.quantity, warehouseDTO.type)
                      })
           )
  }
}
